packages/web3-react/src/hooks/useConnectorMathWallet.ts

```diff
@@ -52,11 +56,12 @@ export const useConnectorMathWallet = (): ConnectorHookResult => {
 
     if (hasInjected() && isMathWalletProvider()) {
       await disconnect();
-      activate(injected);
+      await activate(injected);
+      onConnect?.();
```

Comment:
There are really a lot of duplicates 🤔

Reply:
Reducing duplicates is one of the further goals during reef-knot development.